CMakeLists.txt

```diff
@@ -68,8 +68,8 @@ if(NOT FREECAD_LIBPACK_USE OR FREECAD_LIBPACK_CHECKFILE_CLBUNDLER)
         # OPENGL_gl_LIBRARY empty that results into linker errors
         SetupOpenGL()
     endif(BUILD_GUI)
-    SetupSalomeSMESH()
     SetupBoost()
```

Comment:
Possibly this line re-order is causing the issue.  I do not see where the specific requirement for Boost 1.85.0 is made.  The dependencies between the failed build and recent successful builds on `main` match, so the issue is not due to a dependency upgrade.

SMESH is built with specific versions of Boost as a requirement.  The latest build of SMESH requires Boost 1.84.0 as a dependency within conda-forge.  (ref: [smesh-feedstock](https://github.com/conda-forge/smesh-feedstock/commits/main/))

Reply:
I made the reordering to try to fix this issue (which it did not do, obviously).